app/src/main/java/space/taran/arknavigator/ui/fragments/GalleryFragment.kt

```diff
@@ -191,6 +201,45 @@ class GalleryFragment : MvpAppCompatFragment(), GalleryView {
         startActivity(Intent.createChooser(intent, "Share the link with:"))
     }
 
+    override fun showInfoAlert(path: Path, resourceMeta: ResourceMeta) {
+
+        val dialogBuilder: AlertDialog.Builder =
+            AlertDialog.Builder(requireContext())
+        val dialogResourceInfoBinding = DialogResourceInfoBinding.inflate(
+            layoutInflater
+        )
+        dialogBuilder.setView(dialogResourceInfoBinding.root)
+        // common resources for all file type
+        dialogResourceInfoBinding.resourceId.text = String.format(
+            "Id: %d", resourceMeta.id
+        )
+        dialogResourceInfoBinding.resourceName.text =
+            resourceMeta.name
+        dialogResourceInfoBinding.resourcePath.text =
+            path.absolutePathString()
+        dialogResourceInfoBinding.resourceSize.text =
+            FileUtils.byteCountToDisplaySize(resourceMeta.size)
+        // only for video file type
+        if (resourceMeta.kind is ResourceKind.Video) {
+            dialogResourceInfoBinding.resourceResolution.text = String.format(
+                "%dx%d", resourceMeta.kind.height, resourceMeta.kind.width
+            )
+
+            dialogResourceInfoBinding.resourceDuration.text =
+                resourceMeta.kind.duration?.let {
+                    VideoExtraLoader.durationTextCode(
+                        it
+                    )
+                }
+        }
+        val dialog: AlertDialog = dialogBuilder.create()
+        dialogResourceInfoBinding.cardViewInfo.setOnClickListener {
+            dialog.dismiss()
+        }
+        dialog.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+        dialog.show()
```

Comment:
Extract this code in `ui/fragments/dialog/DetailsAlertDialog`
So that `DetailsAlertDialog(path, meta).show()` only remains here